src/algo/heap.rs

```diff
@@ -362,6 +362,74 @@ pub fn pop_heap<Range>(
     pop_heap_by(rng, start, end, |x, y| x < y);
 }
 
+/// Converts heap in range into sorted range wrt cmp.
+///
+/// # Precondition
+///  - `[start, end)` represents valid positions in rng.
+///  - rng at `[start, end)` is a heap.
+///
+/// # Postcondition
+///  - Sorts the elements in rng such that the whole range is in non-decending order.
```

Comment:
sorts the element in rng at `[start, end)` such that whole range at `[start, end)` is in non-decreasing order wrt cmp.